library/x509_crt.c

```diff
@@ -1605,21 +1605,23 @@ int mbedtls_x509_crt_check_extended_key_usage( const mbedtls_x509_crt *crt,
  */
 int mbedtls_x509_crt_is_revoked( const mbedtls_x509_crt *crt, const mbedtls_x509_crl *crl )
 {
-    const mbedtls_x509_crl_entry *cur = &crl->entry;
+    mbedtls_x509_crl_entry **entry, *key;
+    int ret = 0;
 
-    while( cur != NULL && cur->serial.len != 0 )
-    {
-        if( crt->serial.len == cur->serial.len &&
-            memcmp( crt->serial.p, cur->serial.p, crt->serial.len ) == 0 )
-        {
-            if( mbedtls_x509_time_is_past( &cur->revocation_date ) )
-                return( 1 );
-        }
+    key = mbedtls_calloc(1, sizeof( *key ) );
```

Comment:
There is a reason for that: bsearch() requires a "mbedtls_x509_crl_entry **" as first argument. I am not sure how I could pass that starting from a plain "mbedtls_x509_crl_entry". While I can pass &key when key is "mbedtls_x509_crl_entry *" (maybe this entire thing should be written in a comment).
I'd like to come up with a cleaner method, but I am not sure how. The root cause of this issue is that the entries array in the crl object is an array of "mbedtls_x509_crl_entry *". @andresag01 any suggestion ?